### PR TITLE
Fix/order

### PR DESCRIPTION
**Pull request by msinghartinger**
new method to order detectors based on arrival time

### CODE REVIEW

File: pyipn/detector.py [hunk 1]
```diff
@@ -1,10 +1,14 @@
 import numpy as np
 import astropy.constants as constants
 
-
-from .lightcurve import LightCurve
-from .possion_gen import source_poisson_generator, background_poisson_generator
-
+try:
+    from .lightcurve import LightCurve
```
Comment (grburgess) [inline pyipn/detector.py:5]: Why do we need these import checks?

File: pyipn/geometry.py [hunk 2]
```diff
@@ -32,8 +32,8 @@ def get_seperation_angle(self, grb):
         """
 
         return self._skycoord.separation(grb.location.coord).rad
-
-
+    
+#why pointing and location class different
 class Location(object):
```
Comment (grburgess) [inline pyipn/geometry.py:37]: The location is where the spacecraft is in space. The pointing is what direction in the sky the detector is facing.

File: pyipn/geometry.py [hunk 3]
```diff
@@ -47,6 +47,9 @@ def get_light_travel_time(self, other_location):
     def coord(self):
 
         return self._skycoord
+    
+    def get_cartesian_coord(self):
+        return self._skycoord.represent_as(CartesianRepresentation)
```
Comment (grburgess) [inline pyipn/geometry.py:52]: Does this return a copy or the coord itself?

File: pyipn/universe.py [hunk 4]
```diff
@@ -4,11 +4,22 @@
 from astropy.time import Time
 import astropy.units as u
 
-from .effective_area import EffectiveArea
-from .geometry import Pointing, DetectorLocation
-
-from .grb import GRB
-from .detector import Detector
+try:
```
Comment (grburgess) [inline pyipn/universe.py:7]: import checks

File: pyipn/universe.py [hunk 5]
```diff
@@ -67,14 +78,20 @@ def _compute_time_differences(self):
...
+        grb_vec = self._grb.location.get_cartesian_coord().xyz
+        norm_grb_vec = grb_vec/np.linalg.norm(grb_vec, ord =1) #normalized vector towards GRB
+        
         for name, detector in self._detectors.items():
-
-            ltt.append(detector.angular_separation(self._grb).value)
+            
+            #calculate closest distance to wavefront when the GRB reaches the detector
+            #(negative sign for right order)
+            ltt.append(- np.dot(detector.location.get_cartesian_coord().xyz, norm_grb_vec))
```
Comment (grburgess) [inline pyipn/universe.py:92]: I think I see where you are going!

**Review by grburgess — CHANGES_REQUESTED** [review]
Why do we need these import checks? These should be in the repo

File: pyipn/detector.py (hunk 1 above)
Comment (msinghartinger) [reply pyipn/detector.py:5]: I couldn't get these relative imports to run locally. That's an issue on my end but after almost an hour of unsuccessful troubleshooting I settled for this solution.

File: pyipn/geometry.py (hunk 3 above)
Comment (msinghartinger) [reply pyipn/geometry.py:52]: Good question. I believe represent_as() returns a copy but I'm not sure

**Pull request closed without merging**